homeassistant/components/zha/core/registries.py

```diff
@@ -221,6 +238,41 @@ def _matched(self, manufacturer: str, model: str, channels: list) -> list:
             else:
                 matches.append(model in self.models)
 
+        if self.quirk_classes:
+            if callable(self.quirk_classes):
+                matches.append(self.quirk_classes(quirk_class))
+            else:
+                for quirk in self.quirk_classes:
+                    quirk_tok = quirk.split(".")
+                    if len(quirk_tok) != 2:
+                        _LOGGER.error("Invalid quirk class : '%s'", quirk)
+                        continue
```

Comment:
Just to be sure, is this checking that quirk name have 2 levels?
There will be 'deeper' quirks like `zhaquirks.xiaomi.aqara.sensor_switch_aq3.SwitchAQ3B`